components/DiscussionsCard/DiscussionsCard.tsx

```diff
@@ -21,41 +41,121 @@ const DiscussionsCard: React.FC<DiscussionsCardProps> = ({
   userPic,
   timeStamp,
   content,
-  replyClick,
-  editClick,
   deleteClick
 }) => {
-  return (
-    <div className={isMain ? styles.card_body_main : styles.card_body_sub}>
-      <div
-        className={
-          isMain ? styles.card_user_info_main : styles.card_user_info_sub
-        }
-      >
-        <div>
-          <img src={userPic} className={styles.user_pic} />
-          <span className={styles.username}>{username}</span>
+  const [replyMode, setReplyMode] = useState(false)
+  const [editMode, setEditMode] = useState(false)
+  const [replyText, setReplyText] = useState('')
+  const [editText, setEditText] = useState('')
```

Comment:
Whats the difference between replyText and editText? Why are both kinds of states needed?

Reply:
replyText keeps track of the user inputting the reply, and editText keep track of inputs while a post is being edited by user. Here's a screenshot of both of them being edited at the same time:
![image](https://user-images.githubusercontent.com/77421872/193434598-31d0c2e2-7586-485a-ae55-1a24197e72d4.png)